roles/download/tasks/prep_download.yml

```diff
@@ -13,23 +13,35 @@
     image_info_command: "{{ docker_bin_dir }}/docker images -q | xargs -i {{ '{{' }} docker_bin_dir }}/docker inspect -f {% raw %}'{{ '{{' }} if .RepoTags }}{{ '{{' }} join .RepoTags \",\" }}{{ '{{' }} end }}{{ '{{' }} if .RepoDigests }},{{ '{{' }} join .RepoDigests \",\" }}{{ '{{' }} end }}' {% endraw %} {} | tr '\n' ','"
   when: container_manager == 'docker'
 
-- name: prep_download | Set image pull/info command for containerd and crio
+- name: prep_download | Set image pull/info command for containerd
+  set_fact:
+    image_info_command: "{{ bin_dir }}/ctr -n k8s.io images ls | awk '/application/ {print $1}' | grep -v ^sha | tr '\n' ','"
+    image_pull_command: "{{ bin_dir }}/ctr -n k8s.io images pull --platform linux/{{ image_arch }}"
+  when: container_manager == 'containerd'
+
+- name: prep_download | Set image pull/info command for crio
   set_fact:
     image_info_command: "{{ bin_dir }}/crictl images --verbose | awk -F ': ' '/RepoTags|RepoDigests/ {print $2}' | tr '\n' ','"
     image_pull_command: "{{ bin_dir }}/crictl pull"
-  when: container_manager in ['crio' ,'containerd']
+  when: container_manager == 'crio'
 
 - name: prep_download | Set image pull/info command for docker on localhost
   set_fact:
     image_pull_command_on_localhost: "{{ docker_bin_dir }}/docker pull"
     image_info_command_on_localhost: "{{ docker_bin_dir }}/docker images"
   when: container_manager_on_localhost == 'docker'
 
-- name: prep_download | Set image pull/info command for containerd and crio on localhost
+- name: prep_download | Set image pull/info command for containerd on localhost
+  set_fact:
+    image_info_command_on_localhost: "{{ bin_dir }}/ctr -n k8s.io images ls | awk '/application/ {print $1}' | grep -v ^sha | tr '\n' ','"
+    image_pull_command_on_localhost: "{{ bin_dir }}/ctr -n k8s.io images pull --platform linux/{{ image_arch }}"
```

Comment:
`image_info_command_on_localhost` and `image_pull_command_on_localhost` are the same as `image_info_command` and `image_pull_command`.
We could define
```
image_info_command_on_localhost: "{{ image_info_command }}"
image_pull_command_on_localhost" "{{ image_pull_command }}"
```
but Kubespray already contained such duplicated definitions for crio.
It is fine to go with current change at this time.

Reply:
The `download_on_localhost` seems to be broken currently for anything but docker so this would need further attention in a subsequent PR(s).